c10/Stream.h

```diff
@@ -79,6 +111,15 @@ C10_API std::ostream& operator<<(std::ostream& stream, const Stream& s);
 
 } // namespace c10
 
+namespace std {
+  template <>
+  struct hash<c10::Stream> {
```

Comment:
nit: final

And btw, thanks for making other unrelated classes final in this diff.